File: frontend/src/types/recommendations.ts
export interface RecommendationOut {
  id: string
  title: string
  tmdb_id: number
  poster_url: string | null
  reason: string
  themes: string[]
  status: 'active' | 'dismissed'
  created_at: string
}

export interface GenerateRecommendationsResponse {
  recommendations: RecommendationOut[]
  generated_count: number
}

File: backend/app/schemas/recommendations.py
from pydantic import BaseModel


class RecommendationOut(BaseModel):
    id: str
    title: str
    tmdb_id: int
    poster_url: str | None
    reason: str
    themes: list[str]
    status: str
    created_at: str


class RecommendationsListResponse(BaseModel):
    recommendations: list[RecommendationOut]


class GenerateRecommendationsResponse(BaseModel):
    recommendations: list[RecommendationOut]
    generated_count: int
